File: app/src/main/java/com/checkin/app/checkin/Cook/CookRepository.java
package com.checkin.app.checkin.Cook;

import android.app.Application;
import android.content.Context;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;

import com.checkin.app.checkin.Cook.Model.CookTableModel;
import com.checkin.app.checkin.data.network.ApiClient;
import com.checkin.app.checkin.data.network.ApiResponse;
import com.checkin.app.checkin.data.BaseRepository;
import com.checkin.app.checkin.data.resource.NetworkBoundResource;
import com.checkin.app.checkin.data.resource.Resource;
import com.checkin.app.checkin.data.network.RetrofitLiveData;
import com.checkin.app.checkin.data.network.WebApiService;

import java.util.List;

/**
 * Created by Shivansh Saini on 24/01/2019.
 */

public class CookRepository extends BaseRepository {
    private static CookRepository INSTANCE;
    private final WebApiService mWebService;

    private CookRepository(Context context) {
        mWebService = ApiClient.Companion.getApiService(context);
    }

    public static CookRepository getInstance(Application application) {
        if (INSTANCE == null) {
            synchronized (CookRepository.class) {
                if (INSTANCE == null) {
                    INSTANCE = new CookRepository(application.getApplicationContext());
                }
            }
        }
        return INSTANCE;
    }

    public LiveData<Resource<List<CookTableModel>>> getActiveTables(long shopId) {
        return new NetworkBoundResource<List<CookTableModel>, List<CookTableModel>>() {
            @Override
            protected boolean shouldUseLocalDb() {
                return false;
            }

            @NonNull
            @Override
            protected LiveData<ApiResponse<List<CookTableModel>>> createCall() {
                return new RetrofitLiveData<>(mWebService.getCookActiveTables(shopId));
            }

            @Override
            protected void saveCallResult(List<CookTableModel> data) {

            }
        }.getAsLiveData();
    }
}

File: app/src/main/java/com/checkin/app/checkin/Cook/CookRepository.kt
package com.checkin.app.checkin.Cook

import android.app.Application
import android.content.Context
import androidx.lifecycle.LiveData
import com.checkin.app.checkin.Cook.Model.CookTableModel
import com.checkin.app.checkin.data.BaseRepository
import com.checkin.app.checkin.data.db.ObjectBoxInstanceLiveData
import com.checkin.app.checkin.data.db.dbStore
import com.checkin.app.checkin.data.network.ApiClient.Companion.getApiService
import com.checkin.app.checkin.data.network.ApiResponse
import com.checkin.app.checkin.data.network.RetrofitLiveData
import com.checkin.app.checkin.data.network.WebApiService
import com.checkin.app.checkin.data.resource.NetworkBoundResource
import com.checkin.app.checkin.data.resource.Resource
import com.checkin.app.checkin.menu.MenuRepository
import com.checkin.app.checkin.menu.models.MenuModel
import com.checkin.app.checkin.menu.models.MenuModel_
import com.checkin.app.checkin.menu.models.NewOrderModel
import com.checkin.app.checkin.utility.SingletonHolder

/**
 * Created by Shivansh Saini on 24/01/2019.
 */
class CookRepository private constructor(context: Context) : BaseRepository() {
    private val mWebService: WebApiService = getApiService(context)
    private val cookox by dbStore<CookTableModel>()
    fun getActiveTables(shopId: Long): LiveData<Resource<List<CookTableModel>>> {
        return object : NetworkBoundResource<List<CookTableModel>, List<CookTableModel>>()  {
            override fun shouldUseLocalDb(): Boolean =true





            override fun createCall(): LiveData<ApiResponse<List<CookTableModel>>> {
                return RetrofitLiveData(mWebService.getCookActiveTables(shopId))
            }


        }.asLiveData
    }

    companion object : SingletonHolder<CookRepository, Application>({ CookRepository(it.applicationContext) })

}